zebra-rpc/src/methods/tests/vectors.rs

```diff
@@ -75,3 +75,51 @@ async fn rpc_getblock() {
         assert_eq!(get_block.data, block.into());
     }
 }
+
+#[tokio::test]
+async fn rpc_getbestblockhash() {
+    zebra_test::init();
+
+    // Put `NUMBER_OF_BLOCKS` blocks in a vector
+    let blocks: Vec<Arc<Block>> = zebra_test::vectors::MAINNET_BLOCKS
+        .range(0..=NUMBER_OF_BLOCKS)
+        .map(|(_, block_bytes)| block_bytes.zcash_deserialize_into::<Arc<Block>>().unwrap())
+        .collect();
+
+    // Get the hash of the block at the tip using hardcoded block tip bytes.
+    // We want to test the RPC response is equal to this hash
+    let tip_block: Block = zebra_test::vectors::BLOCK_MAINNET_10_BYTES
+        .zcash_deserialize_into()
+        .unwrap();
+    let tip_block_hash = tip_block.hash();
+
+    // Get a mempool handle
+    let mempool: MockService<_, _, _, BoxError> = MockService::build().for_unit_tests();
+    // Create a populated state service, the tip will be in `NUMBER_OF_BLOCKS`.
+    let state = zebra_state::populated_state(blocks.clone(), Network::Mainnet).await;
+
+    // Init RPC
+    let rpc = RpcImpl {
+        app_version: "Zebra version test".to_string(),
+        mempool: Buffer::new(mempool.clone(), 1),
+        state,
+    };
+
+    // Get the tip hash using RPC method `get_best_block_hash`
+    let get_best_block_hash = rpc
+        .get_best_block_hash()
+        .await
+        .expect("We should have a GetBestBlockHash struct");
+    let response = get_best_block_hash.hex.as_ref().to_vec();
+    let tip_block_hash_bytes = hex::decode(tip_block_hash.to_string()).unwrap();
+
+    // Check if response is equal to block 10 hash.
+    assert_eq!(
+        response,
+        tip_block_hash_bytes
+            .iter()
+            .rev()
+            .cloned()
+            .collect::<Vec::<u8>>()
+    );
```

Comment:
I think you can compare the raw byte slices like this.
(Or you can compare the serialized data if you'd like.)

```suggestion
    let response = get_best_block_hash.hex.as_ref();
    let tip_block_hash_bytes = &tip_block_hash.0;

    // Check if response is equal to block 10 hash.
    assert_eq!(response, tip_block_hash_bytes);
```

Reply:
nice one. https://github.com/ZcashFoundation/zebra/pull/3754/commits/a62f3a0ab6e92f8afe09adbaafef4a440ff7697f